sdk/storage/Azure.Storage.Blobs.ChangeFeed/src/BlobChangeFeedAsyncPageable.cs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Azure.Storage.Blobs.ChangeFeed.Models;
+
+namespace Azure.Storage.Blobs.ChangeFeed
+{
+    /// <summary>
+    /// BlobChangeFeedPagableAsync.
+    /// </summary>
+    internal class BlobChangeFeedAsyncPageable : AsyncPageable<BlobChangeFeedEvent>
+    {
+        private readonly ChangeFeedFactory _changeFeedFactory;
+        private readonly DateTimeOffset? _startTime;
+        private readonly DateTimeOffset? _endTime;
+        private readonly string _continuation;
+
+        /// <summary>
+        /// Internal constructor.
+        /// </summary>
+        internal BlobChangeFeedAsyncPageable(
+            BlobServiceClient blobServiceClient,
+            DateTimeOffset? startTime = default,
+            DateTimeOffset? endTime = default)
+        {
+            _changeFeedFactory = new ChangeFeedFactory(blobServiceClient);
+            _startTime = startTime;
+            _endTime = endTime;
+        }
+
+        internal BlobChangeFeedAsyncPageable(
+            BlobServiceClient blobServiceClient,
+            string continuation)
+        {
+            _changeFeedFactory = new ChangeFeedFactory(blobServiceClient);
+            _continuation = continuation;
+        }
+
+        /// <summary>
+        /// Returns <see cref="BlobChangeFeedEvent"/>s as Pages.
+        /// </summary>
+        /// <param name="continuationToken">
+        /// Throws an <see cref="ArgumentException"/>.  To use contination, call
+        /// <see cref="BlobChangeFeedClient.GetChangesAsync(string)"/>.
+        /// </param>
+        /// <param name="pageSizeHint">
+        /// Page size.
+        /// </param>
+        /// <returns>
+        /// <see cref="IAsyncEnumerable{Page}"/>.
+        /// </returns>
+        public override async IAsyncEnumerable<Page<BlobChangeFeedEvent>> AsPages(
+            string continuationToken = null,
+            int? pageSizeHint = null)
+        {
+            if (continuationToken != null)
+            {
+                throw new ArgumentException($"Continuation not supported.  Use BlobChangeFeedClient.GetChangesAsync(string) instead");
```

Comment:
Because we don't actually use the continuationToken to retrieve the next page.  If the user were to pass it in here, and we actually used it, we would have to re-initialize the change feed.  Originally, I was going to just ignore the continuationToken if the user passed it in, but we made the decision as the team that we should throw.